app/components/QrCodeScanner/QrCodeScanner.jsx

```diff
@@ -29,15 +40,22 @@ export default class QrCodeScanner extends Component<Props> {
       callback(content, this.stopScanner.bind(this))
     })
 
-    Instascan.Camera.getCameras()
+    // since the browser halts while getting usermedia
+    // let the loading animation finish one round and only then continue
+    this.setState({ loading: true })
+    new Promise(resolve => setTimeout(resolve, 900))
```

Comment:
**Why force a timeout?**
Without it the loading animation hangs. Look:
![ezgif com-video-to-gif 3](https://user-images.githubusercontent.com/486954/49511534-19699c00-f894-11e8-90c6-3bc78159b67b.gif)

**Why does the animation hang?**
Because getting camera access stalls the browser. I validated this with the Chrome profiler. For about a second once `getUserMedia` is triggered the browser does absolutely nothing! I experimented with GPU layering but there doesn't seem to be a way around it.

**Why 900ms?**
That's the time it takes for the animation to complete a cycle,  and so this way, the animation hang isn't noticeable.

Reply:
Ha, I just asked about this on another PR. Perhaps add a variable name to explain or a comment

Reply:
@ranbena this is awesome just curious is there a fixed amount of time it takes for camera to render? Is that dependent on hardware?

Reply:
That's a good question. I found no info on the web. Any suggestion/reference would be welcome.

If the render time takes longer on other systems, it would result in a longer time between the completed animation cycle and the actual preview coming up. I think it's the best we got at this time.